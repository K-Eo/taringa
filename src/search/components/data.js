export const user = {
  id: '27s73',
  created: 1513920502005,
  type: 'user',
  tid: 924744,
  username: 'supr159',
  firstname: 'Diego',
  lastname: 'Lopez',
  birthday: '1992-1-18',
  avatar:
    'https://a21.t26.net/taringa/avatares/4/6/6/F/D/5/supr159/120x120_B7B.jpg',
  site: '',
  gender: 'm',
  country: 'MX',
  message: '',
  subscribedToDefaultChannels: true,
  suspended: false,
  stories: 11,
  followers: 32,
  following: 0,
  rank: 'User',
  latestStory: '2018-01-17T03:07:26.000Z',
  state: {
    following: false,
  },
}
