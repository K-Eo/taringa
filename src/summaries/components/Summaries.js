import _ from 'lodash'
import React from 'react'
import classNames from 'classnames'

import { ITEM_BIG, ITEM_MEDIUM, ITEM_SMALL } from '../../settings/constants'
import Summary from './Summary'

const placeholdersByItemSize = {
  [ITEM_BIG]: 3,
  [ITEM_MEDIUM]: 9,
  [ITEM_SMALL]: 19,
}

const Summaries = ({ items, onRetry, size, status }) => {
  const isFetching = status === 'fetching'
  const hasFailure = status === 'failure'
  const itemsLength = items.length

  const makePlaceholders = () => {
    const effectiveCount = itemsLength > 0 ? 1 : placeholdersByItemSize[size]

    return _.times(effectiveCount, index => (
      <Summary key={index} isPlaceholder size={size} />
    ))
  }

  const classes = classNames('Summaries', {
    'item-big': size === ITEM_BIG,
    'item-medium': size === ITEM_MEDIUM,
    'item-small': size === ITEM_SMALL,
  })

  return (
    <div className={classes}>
      {(!hasFailure || itemsLength > 0) && (
        <div className="card">
          <ul className="list-group list-group-flush">
            {items.map(item => <Summary key={item.id} {...item} />)}

            {isFetching && makePlaceholders()}
          </ul>
        </div>
      )}

      {hasFailure && (
        <div className="my-4 text-center">
          <div className="my-3 text-danger">
            ¡Ratas! Parece que no estás conectado a internet. Refresca la página
            o pulsa el siguiente botón.
          </div>

          <button className="btn btn-outline-primary btn-sm" onClick={onRetry}>
            Volver a intentar
          </button>
        </div>
      )}
    </div>
  )
}

Summaries.defaultProps = {
  items: [],
  onRetry: () => {},
  size: ITEM_MEDIUM,
  status: 'success',
}

export default Summaries
