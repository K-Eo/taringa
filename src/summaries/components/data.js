export const image = {
  id: 'zbmsb',
  created: 1513920502005,
  edited: null,
  title: '',
  nsfw: false,
  classic: {
    tid: 'cH3t9',
    type: 'shout',
  },
  tags: ['image', 'classic'],
  slug: 'si-no-das-like-no-tendras-pelirojas-en-100-anos_zbmsb',
  type: 'story:summary',
  summary: {
    link: null,
    images: {
      slice: [
        {
          type: 'image',
          url: 'https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/D4C.jpg',
          width: 779,
          height: 1026,
          format: 'jpeg',
        },
      ],
      amount: 1,
    },
    excerpt: 'Si no das like no tendras pelirojas en 100 años',
    textLength: 47,
  },
  channel: {
    type: 'channel:summary',
    id: 'ixoir',
    created: '2016-08-28T21:54:19.000Z',
    name: 'user-Fraudier',
    title: 'user-Fraudier',
    channelType: 'user_feed',
    thumbnail: '',
    background: '',
    description: '',
    category: 'unknown/unknown',
    owner: {
      id: 'ixoiq',
      created: '2016-08-28T21:54:19.000Z',
      type: 'user',
      tid: 27914320,
      username: 'Fraudier',
      firstname: 'Fraudier',
      lastname: 'ㅤ',
      birthday: '1910-9-19',
      avatar:
        'https://a12.t26.net/taringa/avatares/5/E/3/6/4/7/Fraudier/120x120_493.jpg',
      site: 'http://www.fraudierdrogbaa.com',
      gender: 'm',
      country: 'VG',
      message: 'El soldado de fermas',
      migrated: [],
      subscribedToDefaultChannels: false,
    },
    subscribers: 0,
    stories: 1120,
    latestStory: '2018-02-06T04:19:50.000Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: 'ixoiq',
    username: 'Fraudier',
    type: 'user:summary',
    firstname: 'Fraudier',
    lastname: 'ㅤ',
    avatar:
      'https://a12.t26.net/taringa/avatares/5/E/3/6/4/7/Fraudier/120x120_493.jpg',
    gender: 'm',
    country: 'VG',
  },
  upvotes: 2905,
  downvotes: 0,
  shares: 1,
  comments: 78,
  state: {},
  brandSafe: false,
  listViewSafe: false,
}

export const text = {
  id: 'zcvzz',
  created: 1513920502005,
  edited: null,
  title: '',
  nsfw: false,
  classic: {
    tid: 'cHDmR',
    type: 'shout',
  },
  tags: ['text', 'classic'],
  slug:
    'el-de-kapanga-echo-a-un-pibe-del-recital-por-tener-una-remera-con-la-b_zcvzz',
  type: 'story:summary',
  summary: {
    link: null,
    images: {
      slice: [],
      amount: 0,
    },
    excerpt:
      'El de Kapanga echó a un pibe del recital por tener una remera con la bandera inglesa, temiendo que pudiera incitar a su público a...',
    textLength: 167,
  },
  channel: {
    type: 'channel:summary',
    id: 'fswgi',
    created: '2016-08-17T22:02:01.000Z',
    name: 'user-Fixie-',
    title: 'user-Fixie-',
    channelType: 'user_feed',
    thumbnail: '',
    background: '',
    description: '',
    category: 'unknown/unknown',
    owner: {
      id: 'fswgh',
      created: '2016-08-17T22:02:01.000Z',
      type: 'user',
      tid: 24379422,
      username: 'Fixie-',
      firstname: 'Idiota',
      lastname: '.',
      birthday: '1998-4-4',
      avatar:
        'https://a21.t26.net/taringa/avatares/9/E/2/C/1/5/Fixie-/120x120_A50.jpg',
      site: 'http://untitled-pibe.tumblr.com/',
      gender: 'm',
      country: 'IT',
      message: 'zisis a bath tub',
      migrated: [],
      subscribedToDefaultChannels: false,
    },
    subscribers: 0,
    stories: 5978,
    latestStory: '2018-02-07T22:04:49.000Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: 'fswgh',
    username: 'Fixie-',
    type: 'user:summary',
    firstname: 'Idiota',
    lastname: '.',
    avatar:
      'https://a21.t26.net/taringa/avatares/9/E/2/C/1/5/Fixie-/120x120_A50.jpg',
    gender: 'm',
    country: 'IT',
  },
  upvotes: 41,
  downvotes: 0,
  shares: 0,
  comments: 0,
  state: {},
  brandSafe: false,
  listViewSafe: false,
}

export const article = {
  id: 'zcwgr',
  created: 1513920502005,
  edited: '2018-02-07T21:39:24.723Z',
  title: 'consejos para sobrevivir al ataque de un animal salvaje',
  nsfw: false,
  classic: {
    tid: 20125784,
    type: 'post',
    images: {
      '1:1': {
        type: 'image',
        url: 'https://t3.kn3.net/taringa/4/B/B/5/4/A/arcangelg80/01A.jpg',
        width: 526,
        height: 526,
        format: 'jpeg',
      },
      '4:3': {
        type: 'image',
        url: 'https://t1.kn3.net/taringa/F/0/C/8/7/C/arcangelg80/42E.jpg',
        width: 526,
        height: 395,
        format: 'jpeg',
      },
    },
  },
  tags: ['article', 'classic'],
  slug: 'consejos-para-sobrevivir-al-ataque-de-un-animal-salvaje_zcwgr',
  type: 'story:summary',
  summary: {
    link: null,
    images: {
      slice: [
        {
          type: 'image',
          url: 'https://t1.kn3.net/taringa/F/0/C/8/7/C/arcangelg80/42E.jpg',
          width: 526,
          height: 395,
          format: 'jpeg',
        },
        {
          type: 'image',
          url: 'https://k30.kn3.net/taringa/1/C/B/6/D/0/MaestroRoshiII/01B.gif',
          width: 650,
          height: 200,
          format: 'gif',
        },
        {
          type: 'image',
          url: 'https://k50.kn3.net/taringa/6/5/6/9/3/5/868.gif',
          width: 435,
          height: 250,
          format: 'gif',
        },
      ],
      amount: 10,
    },
    excerpt:
      'https://kn3.net/MaestroRoshiII/30-1-C-B-6-D-0-01B-GIF.html\nSi bien es cierto que es muy raro ser atacado por animales salvajes, por otro ...',
    textLength: 3926,
  },
  channel: {
    type: 'channel:summary',
    id: '22e',
    created: '2016-08-05T17:55:08.000Z',
    name: 'hazlo_tu_mismo',
    title: 'Hazlo tú mismo',
    channelType: 'public',
    thumbnail: 'https://k60.kn3.net/taringa/E/5/8/7/7/B/FB6.png',
    background: 'https://k60.kn3.net/taringa/F/1/F/6/F/8/E8D.png',
    description: 'Hazlo tú mismo',
    category: 'diy/tutorials',
    owner: {
      id: '17t',
      created: '2016-07-29T16:57:34.000Z',
      type: 'user',
      tid: 2987,
      username: 'Taringa',
      firstname: 'Taringa!',
      lastname: 'Inteligencia Colectiva',
      birthday: '1994-4-14',
      avatar:
        'https://a11.t26.net/taringa/avatares/3/D/6/2/3/8/Taringa/120x120_943.jpg',
      site: 'http://www.taringa.net',
      gender: 'm',
      country: 'AR',
      message: 'Inteligencia Colectiva',
      migrated: ['followers', 'classicCategories'],
      subscribedToDefaultChannels: true,
    },
    subscribers: 319050,
    stories: 45398,
    latestStory: '2018-02-07T22:35:28.568Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: 'a7716',
    username: 'arcangelg80',
    type: 'user:summary',
    firstname: 'angel',
    lastname: 'gonzalez',
    avatar:
      'https://a21.t26.net/taringa/avatares/A/4/6/B/2/0/arcangelg80/120x120_CDD.jpg',
    gender: 'm',
    country: 'VE',
  },
  upvotes: 9,
  downvotes: 0,
  shares: 0,
  comments: 14,
  state: {},
  brandSafe: true,
  listViewSafe: false,
}

export const link = {
  id: 'zcxyl',
  created: 1513920502005,
  edited: null,
  title: 'Godoy Cruz-Lanús, Superliga: juegan en Mendoza un partido pendiente',
  nsfw: false,
  tags: ['link', 'brainiac'],
  slug:
    'godoy-cruz-lanus-superliga-juegan-en-mendoza-un-partido-pendiente_zcxyl',
  type: 'story:summary',
  summary: {
    link: {
      type: 'link',
      url: 'https://www.lanacion.com.ar/2107299-godoy-cruz-lanus-superliga',
      resourceType: 'html',
      originalUrl:
        'http://www.lanacion.com.ar/2107299-godoy-cruz-lanus-superliga?utm_source=taringa&utm_campaign=rss',
      provider: {
        type: 'provider',
        url: 'https://www.lanacion.com.ar',
        name: 'Lanacion',
        display: 'www.lanacion.com.ar',
      },
      favicon: 'https://k50.kn3.net/taringa/F/4/C/5/E/8/3D0.png',
      title:
        'Godoy Cruz-Lanús, Superliga: juegan en Mendoza un partido pendiente',
      description:
        "Godoy Cruz y Lanús juegan hoy en Mendoza el partido que tienen pendiente de la décima fecha de la Superliga del fútbol argentino, postergado por la participación del 'Granate´ en la final de la Copa Libertadores 2017.",
      images: [
        {
          type: 'image',
          url: 'https://k50.kn3.net/taringa/9/F/2/5/E/D/EB2.jpg',
          width: 900,
          height: 599,
          format: 'jpeg',
        },
      ],
      media: null,
      authors: [],
    },
    images: {
      slice: [],
      amount: 0,
    },
    excerpt: '',
    textLength: 0,
  },
  channel: {
    type: 'channel:summary',
    id: '1jq',
    created: '2016-08-04T21:06:31.000Z',
    name: 'deportes',
    title: 'Deportes',
    channelType: 'public',
    thumbnail: 'https://k60.kn3.net/taringa/9/E/E/B/3/0/589.png',
    background: 'https://k60.kn3.net/taringa/C/7/E/1/C/A/E4C.jpg',
    description: 'Deportes',
    category: 'sports/other',
    owner: {
      id: '17t',
      created: '2016-07-29T16:57:34.000Z',
      type: 'user',
      tid: 2987,
      username: 'Taringa',
      firstname: 'Taringa!',
      lastname: 'Inteligencia Colectiva',
      birthday: '1994-4-14',
      avatar:
        'https://a11.t26.net/taringa/avatares/3/D/6/2/3/8/Taringa/120x120_943.jpg',
      site: 'http://www.taringa.net',
      gender: 'm',
      country: 'AR',
      message: 'Inteligencia Colectiva',
      migrated: ['followers', 'classicCategories'],
      subscribedToDefaultChannels: true,
    },
    subscribers: 317511,
    stories: 275050,
    latestStory: '2018-02-07T23:25:13.000Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: '1kphe',
    username: 'lanacion',
    type: 'user:summary',
    firstname: 'La Nación',
    lastname: 'RSS',
    avatar:
      'https://a21.t26.net/taringa/avatares/D/0/8/9/1/4/lanacion/120x120_96C.jpg',
    gender: 'm',
    country: 'AR',
  },
  upvotes: 0,
  downvotes: 0,
  shares: 0,
  comments: 0,
  state: {},
  brandSafe: false,
  listViewSafe: false,
}
