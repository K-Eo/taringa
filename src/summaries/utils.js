import { humanizeNum } from '../Utils'

const scaleThumbnail = image => {
  if (!image) {
    return null
  }

  const { height, url, width } = image
  const minSize = 78
  let nextSize = ''

  if (width < height) {
    nextSize = `${minSize}x${minSize * 2}`
  } else if (height < width) {
    nextSize = `${minSize * 2}x${minSize}`
  } else {
    nextSize = `${minSize}x${minSize}`
  }

  if (url.indexOf('kn3') === -1) {
    return url
  }

  return url.replace(/(\w+?)(\.\w+?)$/g, `${nextSize}_$1$2`)
}

const getThumbnail = (story = {}) => {
  let image = null

  if (story.summary.images.amount > 0) {
    image = story.summary.images.slice[0]
  } else if (
    story.summary.link !== null &&
    story.summary.link.images.length > 0
  ) {
    image = story.summary.link.images[0]
  }

  return scaleThumbnail(image)
}

const getTitle = (story = {}) => {
  if ('title' in story && story.title.length > 0) {
    return story.title
  } else {
    return story.summary.excerpt
  }
}

const getIcon = (story = {}) => {
  const [kind] = story.tags

  switch (kind) {
    case 'image':
      return 'fa fa-image'
    case 'link':
      return 'fa fa-link'
    case 'article':
      return 'fa fa-file-alt'
    case 'text':
      return 'fa fa-comment'
    default:
      return 'fa fa-question'
  }
}

const getPreview = (story = {}) => {
  const [kind] = story.tags

  switch (kind) {
    case 'image':
    case 'article':
      if (story.summary.images.amount > 0) {
        return {
          kind: 'image',
          content: story.summary.images.slice[0].url,
        }
      }

      return null
    case 'text':
      return {
        kind: 'text',
        content: story.summary.excerpt,
      }
    case 'link':
      return {
        kind: 'text',
        content: story.summary.link.description,
      }
    default:
      return null
  }
}

const getChannel = story => {
  const { channel } = story
  const result = { channelName: channel.title, channel: channel.name }

  if (channel.channelType === 'user_feed') {
    result.channelName = null
  }

  return result
}

export const normalizeStory = story => {
  return {
    comments: story.comments,
    created: story.created,
    icon: getIcon(story),
    id: story.id,
    owner: story.owner.username,
    preview: getPreview(story),
    score: humanizeNum(story.upvotes - story.downvotes),
    shares: story.shares,
    slug: story.slug,
    thumbnail: getThumbnail(story),
    title: getTitle(story),
    ...getChannel(story),
  }
}
