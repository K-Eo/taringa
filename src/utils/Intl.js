import buildFormatter from 'react-timeago/lib/formatters/buildFormatter'

const ESFormatter = buildFormatter({
  prefixAgo: 'hace',
  prefixFromNow: 'dentro de',
  suffixAgo: '',
  suffixFromNow: 'justo ahora',
  seconds: '%d segundos',
  minute: '1 minuto',
  minutes: '%d minutos',
  hour: '1 hora',
  hours: '%d horas',
  day: '1 día',
  days: '%d días',
  month: '1 mes',
  months: '%d meses',
  year: '1 año',
  years: '%d años',
})

const ESShortFormatter = buildFormatter({
  prefixAgo: '\u2219',
  prefixFromNow: '',
  suffixAgo: '',
  suffixFromNow: '',
  seconds: '%d s',
  minute: '1 m',
  minutes: '%d m',
  hour: '1 h',
  hours: '%d h',
  day: '1 d',
  days: '%d d',
  month: '1 m',
  months: '%d m',
  year: '1 a',
  years: '%d a',
})

Intl.ES = function() {
  return ESFormatter
}

Intl.ESShort = function() {
  return ESShortFormatter
}
