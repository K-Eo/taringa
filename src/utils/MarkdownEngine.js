import _ from 'lodash'
import React from 'react'
import ReactMarkdown from 'react-markdown'
import queryString from 'query-string'
import YouTube from 'react-youtube'

class MarkdownEngine {
  constructor() {
    this.mapEntities = this.mapEntities.bind(this)
  }

  processHashtags(body) {
    return body.replace(/<hashtag tag="(.+?)" \/>/g, '[#$1](/hashtags/$1)')
  }

  processMentions(body) {
    return body.replace(/<mention user="(.+?)" \/>/g, '[@$1](/user/$1)')
  }

  preRender(body) {
    return _.reduce(
      [this.processHashtags, this.processMentions],
      (md, processor) => {
        if (processor) {
          return processor(md)
        }

        return md
      },
      body
    )
  }

  getVideoId(url) {
    const { query } = queryString.parseUrl(url)
    return query.v
  }

  mapLinkEntity(entity, i) {
    switch (entity.resourceType) {
      case 'video.other':
        const id = this.getVideoId(entity.url)
        if (id) {
          return (
            <div className="text-center" key={i}>
              <YouTube key={i} videoId={id} />
            </div>
          )
        }
        return null
      default:
        return null
    }
  }

  mapEntities(entity, i) {
    switch (entity.type) {
      case 'markdown':
        return <ReactMarkdown key={i} source={this.preRender(entity.body)} />
      case 'image':
        return (
          <img
            alt={entity.url}
            height={entity.height}
            key={i}
            src={entity.url}
            width={entity.width}
          />
        )
      case 'html':
        return <div key={i} dangerouslySetInnerHTML={{ __html: entity.body }} />
      case 'link':
        return this.mapLinkEntity(entity, i)
      default:
        return null
    }
  }

  render(content) {
    return _.map(content, this.mapEntities)
  }

  static Render(content) {
    const engine = new MarkdownEngine()
    return engine.render(content)
  }
}

export default MarkdownEngine
