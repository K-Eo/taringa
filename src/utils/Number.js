import _ from 'lodash'

export function pluralize(singular, plural = null, onlyWord = false) {
  let word = ''

  if (this && this === 1) {
    word = singular
  } else {
    word = plural || `${singular}s`
  }

  if (onlyWord) {
    return word
  }

  return `${this} ${word}`
}

export function humanize(decimals = 1) {
  let original = this

  let number = Math.abs(this)
  decimals = Math.pow(10, decimals)

  const abbreviations = ['k', 'm', 'b', 't']

  for (let i = abbreviations.length - 1; i >= 0; i--) {
    const size = Math.pow(10, (i + 1) * 3)

    if (size <= number) {
      number = Math.round(number * decimals / size) / decimals

      if (number === 1000 && i < abbreviations.length - 1) {
        number = 1
        i++
      }

      number += abbreviations[i]

      break
    }
  }

  if (original < 0) {
    return `-${number.toString()}`
  }

  return number.toString()
}

_.assign(Number.prototype, {
  humanize,
  pluralize,
})
