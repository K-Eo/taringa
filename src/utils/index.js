import './String.js'
import './Number.js'
import './Intl.js'
