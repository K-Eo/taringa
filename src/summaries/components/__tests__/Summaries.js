import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Summaries from '../Summaries'
import { image } from '../data'
import { normalizeStory } from '../../utils'

describe('Summaries', () => {
  beforeEach(() => {
    Date.now = jest.fn(() => 1513924592705)
  })

  it('renders default', () => {
    const wrapper = shallow(<Summaries />)

    expect(wrapper).toMatchSnapshot()
  })

  it('changes placeholder count', () => {
    const wrapper = shallow(
      <Summaries status="fetching" placeholderCount={10} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('renders error', () => {
    const wrapper = shallow(<Summaries status="failure" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders summaries', () => {
    const wrapper = shallow(
      <Summaries status="success" items={[normalizeStory(image)]} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('calls onLoad', () => {
    const mock = jest.fn()

    mount(<Summaries onLoad={mock} />)

    expect(mock).toHaveBeenCalled()
  })

  it('calls onRetry when has failure', () => {
    const mock = jest.fn()

    const wrapper = shallow(<Summaries status="failure" onRetry={mock} />)

    wrapper.find('button').simulate('click')

    expect(mock).toHaveBeenCalled()
  })
})
