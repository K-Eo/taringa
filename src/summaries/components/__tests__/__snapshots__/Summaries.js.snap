// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summaries Failure renders error 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="my-4 text-center"
  >
    <div
      className="my-3 text-danger"
    >
      ¡Ratas! Parece que no estás conectado a internet. Refresca la página o pulsa el siguiente botón.
    </div>
    <button
      className="btn btn-outline-primary btn-sm"
      onClick={[Function]}
    >
      Volver a intentar
    </button>
  </div>
</div>
`;

exports[`Summaries Failure renders error whan has items 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        channel="user-Fraudier"
        channelName={null}
        comments={78}
        created={1513920502005}
        icon="far fa-image"
        id="zbmsb"
        isPlaceholder={false}
        key="zbmsb"
        owner="Fraudier"
        preview={
          Object {
            "content": "https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/D4C.jpg",
            "kind": "image",
          }
        }
        score="2.9k"
        shares={1}
        slug="si-no-das-like-no-tendras-pelirojas-en-100-anos_zbmsb"
        thumbnail="https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/78x1026_D4C.jpg"
        title="Si no das like no tendras pelirojas en 100 años"
      />
    </ul>
  </div>
  <div
    className="my-4 text-center"
  >
    <div
      className="my-3 text-danger"
    >
      ¡Ratas! Parece que no estás conectado a internet. Refresca la página o pulsa el siguiente botón.
    </div>
    <button
      className="btn btn-outline-primary btn-sm"
      onClick={[Function]}
    >
      Volver a intentar
    </button>
  </div>
</div>
`;

exports[`Summaries Placeholder renders 1 placeholder when has items and is fetching 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        channel="user-Fraudier"
        channelName={null}
        comments={78}
        created={1513920502005}
        icon="far fa-image"
        id="zbmsb"
        isPlaceholder={false}
        key="zbmsb"
        owner="Fraudier"
        preview={
          Object {
            "content": "https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/D4C.jpg",
            "kind": "image",
          }
        }
        score="2.9k"
        shares={1}
        slug="si-no-das-like-no-tendras-pelirojas-en-100-anos_zbmsb"
        thumbnail="https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/78x1026_D4C.jpg"
        title="Si no das like no tendras pelirojas en 100 años"
      />
      <Summary
        isPlaceholder={true}
        key="0"
        size="settings/ITEM_MEDIUM"
      />
    </ul>
  </div>
</div>
`;

exports[`Summaries Placeholder renders 3 placeholders when size is big 1`] = `
<div
  className="Summaries item-big"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        isPlaceholder={true}
        key="0"
        size="settings/ITEM_BIG"
      />
      <Summary
        isPlaceholder={true}
        key="1"
        size="settings/ITEM_BIG"
      />
      <Summary
        isPlaceholder={true}
        key="2"
        size="settings/ITEM_BIG"
      />
    </ul>
  </div>
</div>
`;

exports[`Summaries Placeholder renders 9 placeholders when size is medium 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        isPlaceholder={true}
        key="0"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="1"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="2"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="3"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="4"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="5"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="6"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="7"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="8"
        size="settings/ITEM_MEDIUM"
      />
    </ul>
  </div>
</div>
`;

exports[`Summaries Placeholder renders 9 placeholders when size is no given nor placeholder count 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        isPlaceholder={true}
        key="0"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="1"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="2"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="3"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="4"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="5"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="6"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="7"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="8"
        size="settings/ITEM_MEDIUM"
      />
    </ul>
  </div>
</div>
`;

exports[`Summaries Placeholder renders 19 placeholders when size is small 1`] = `
<div
  className="Summaries item-small"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        isPlaceholder={true}
        key="0"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="1"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="2"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="3"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="4"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="5"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="6"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="7"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="8"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="9"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="10"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="11"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="12"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="13"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="14"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="15"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="16"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="17"
        size="settings/ITEM_SMALL"
      />
      <Summary
        isPlaceholder={true}
        key="18"
        size="settings/ITEM_SMALL"
      />
    </ul>
  </div>
</div>
`;

exports[`Summaries Placeholder renders given placeholders 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        isPlaceholder={true}
        key="0"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="1"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="2"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="3"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="4"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="5"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="6"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="7"
        size="settings/ITEM_MEDIUM"
      />
      <Summary
        isPlaceholder={true}
        key="8"
        size="settings/ITEM_MEDIUM"
      />
    </ul>
  </div>
</div>
`;

exports[`Summaries renders default 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    />
  </div>
</div>
`;

exports[`Summaries renders summaries 1`] = `
<div
  className="Summaries item-medium"
>
  <div
    className="card"
  >
    <ul
      className="list-group list-group-flush"
    >
      <Summary
        channel="user-Fraudier"
        channelName={null}
        comments={78}
        created={1513920502005}
        icon="far fa-image"
        id="zbmsb"
        isPlaceholder={false}
        key="zbmsb"
        owner="Fraudier"
        preview={
          Object {
            "content": "https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/D4C.jpg",
            "kind": "image",
          }
        }
        score="2.9k"
        shares={1}
        slug="si-no-das-like-no-tendras-pelirojas-en-100-anos_zbmsb"
        thumbnail="https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/78x1026_D4C.jpg"
        title="Si no das like no tendras pelirojas en 100 años"
      />
    </ul>
  </div>
</div>
`;
